snoop/snoop_test.go

```diff
@@ -0,0 +1,163 @@
+// Copyright - Copyleft 2019 Ripx80, All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+package snoop
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	spHeader = []byte{
+		0x73, 0x6E, 0x6F, 0x6F, 0x70, 0x00, 0x00, 0x00,
+		0x00, 0x00, 0x00, 0x02,
+		0x00, 0x00, 0x00, 0x04,
+	}
+
+	pack = []byte{
+		0x00, 0x00, 0x00, 0x2A, 0x00, 0x00, 0x00, 0x2A, 0x00, 0x00, 0x00, 0x44, 0x00, 0x00, 0x00, 0x00, 0x5C, 0xBE, 0xB8, 0x4C, 0x00, 0x0C, 0xB1, 0x47,
+		0x7c, 0x5a, 0x1c, 0x49, 0x3c, 0xd1, 0x1e, 0x65, 0x50, 0x7f, 0xb9, 0xca, 0x08, 0x06, 0x00, 0x01, 0x08, 0x00, 0x06, 0x04, 0x00, 0x01, 0x1e, 0x65,
+		0x50, 0x7f, 0xb9, 0xca, 0x0a, 0x00, 0x33, 0x68, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0a, 0x00, 0x33, 0x01, 0x00, 0x00,
+	}
+)
+
+func OpenHandlePack() (buf []byte, handle *Reader, err error) {
+	buf = make([]byte, len(spHeader)+len(pack))
+	copy(buf, append(spHeader, pack...))
+	handle, err = NewReader(bytes.NewReader(buf))
+	return buf, handle, err
+}
+
+func TestReadHeader(t *testing.T) {
+	_, err := NewReader(bytes.NewReader(spHeader))
+	assert.Nil(t, err)
```

Comment:
Could you please use comparisons instead of assert? This would add a dependency on "github.com/stretchr/testify/assert"

Reply:
Remove testify. Now I used *testing.T.Helper() to get the right line numbers but this requires go version [1.9](https://golang.org/doc/go1.9#test-helper). If that's a problem I will remove the Helper call or do the checks without a extra function.